firefox-ios/Client/Frontend/Browser/Menu/MainMenuViewController.swift

```diff
@@ -0,0 +1,173 @@
+    ) {
+        self.viewModel = viewModel
+        self.windowUUID = windowUUID
+        self.notificationCenter = notificationCenter
+        self.themeManager = themeManager
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View setup & lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        presentationController?.delegate = self
+        sheetPresentationController?.delegate = self
+        scrollView.delegate = self
+
+        setupNotifications(forObserver: self,
+                           observing: [.DynamicFontChanged])
+
+        setupView()
+        listenForThemeChange(view)
+        subscribeToRedux()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        applyTheme()
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        updateModalA11y()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+    }
+
+    // MARK: - View setup
+    private func setupView() { }
+
+    private func updateContent() {
+        contentStackView.removeAllArrangedViews()
+        applyTheme()
+    }
+
+    // MARK: - Redux
+    func subscribeToRedux() {
+        let uuid = windowUUID
+        store.subscribe(self, transform: {
+            $0.select({ appState in
+                return BrowserViewControllerState(appState: appState, uuid: uuid)
+            })
+        })
+    }
+
+    func unsubscribeFromRedux() {
+        store.unsubscribe(self)
+    }
+
+    func newState(state: BrowserViewControllerState) {
+    }
+
+    // MARK: - UX related
+    func applyTheme() {
+        let theme = themeManager.getCurrentTheme(for: windowUUID)
+        view.backgroundColor = theme.colors.layer1
+    }
+
+    // MARK: - Notifications
+    func handleNotifications(_ notification: Notification) { }
+
+    @objc
+    private func closeTapped() { }
+
+    deinit {
+        unsubscribeFromRedux()
+    }
+
+    private func updateModalA11y() {
+        var currentDetent: UISheetPresentationController.Detent.Identifier? = viewModel.getCurrentDetent(
+            for: sheetPresentationController
+        )
+
+        if currentDetent == nil,
+           let sheetPresentationController,
+           let firstDetent = sheetPresentationController.detents.first {
+            if firstDetent == .medium() {
+                currentDetent = .medium
+            } else if firstDetent == .large() {
+                currentDetent = .large
+            }
+        }
+
+        // in iOS 15 modals with a large detent read content underneath the modal in voice over
+        // to prevent this we manually turn this off
```

Comment:
nit: maybe add this comment to top of the method and that we can get rid of this when we drop the support of the version?

I missed this when creating the microsurvey modal, so I will double check if I need to add this to the microsurvey modal as well. Voice over works well on iOS 17 without this check when previously tested.

Reply:
Also I see some similar logic between modals, so maybe we can make future modals conform to a parent modal class, but out of scope and just a thought.